lib/baseFactory.js

```diff
@@ -116,17 +116,20 @@ class BaseFactory {
     list(config) {
         const scanConfig = {
             table: this.table,
-            params: config.params || {},
-            paginate: {
-                count: hoek.reach(config, 'paginate.count', { default: PAGINATE_COUNT }),
-                page: hoek.reach(config, 'paginate.page', { default: PAGINATE_PAGE })
-            }
+            params: hoek.reach(config, 'params', { default: {} })
         };
 
-        if (config.sort) {
+        if (config && config.sort) {
             scanConfig.sort = config.sort;
         }
 
+        if (config && config.paginate) {
+            scanConfig.paginate = {
+                count: hoek.reach(config, 'paginate.count', { default: PAGINATE_COUNT }),
```

Comment:
This hoek.reach behaves weird when paginate.count = undefined.
```
> var config = { paginate: { count: undefined } }
>  hoek.reach(config, 'paginate.count', { default: 5 })
undefined
```
See it's not using the default value when paginate.count's value is `undefined`. We need to take care of that case as well.  need to do `hoek.reach(config, 'paginate.count', { default: PAGINATE_COUNT }) || PAGINATE_COUNT`. Not sure if there is any other way to do it to make it cleaner